### PR TITLE
[docs] refactor algolia docsearch implementation

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Right now, the Expo SDK versions are [hardcoded in the Docsearch configuration](https://github.com/algolia/docsearch-configs/blob/master/configs/expo.json#L7-L14). That means we have to [manually update this](https://github.com/algolia/docsearch-configs/pulls?q=is%3Apr+is%3Aclosed+expo) every time we release a new version. It's bringing some maintenance burden and also provide room for human error. With the release of v37, we deprecated v33 and removed this from the config. But, we didn't remove this from our docs. That resulted in Algolia indexing SDK 33 without a version (by crawling the pages), returning SDK 33 API references in every query.

# How

**Fixing the config maintenance**
This change removes the need for hardcoded versions in the Algolia docsearch config. Instead of writing them there, it adds a new meta tag on every page (like [Bootstrap Table](https://github.com/algolia/docsearch-configs/blob/796e9f7658f84f59a33d985522be13d2a835aa7a/configs/bootstrap-table.json) or [Laravel](https://github.com/algolia/docsearch-configs/blob/796e9f7658f84f59a33d985522be13d2a835aa7a/configs/laravel_nova.json)). With this, we can just publish a new version or deprecate an old one without having to change the docsearch config. We only need to wait for a reindex ([should happen every 24 hours](https://docsearch.algolia.com/docs/faq/#how-often-will-you-crawl-my-website))

```
// for versioned pages (reference)
<meta name="docsearch:version" content="v37.0.0">
<meta name="docsearch:version" content="v35.0.0">

// for non-versioned pages (guides/started)
<meta name="docsearch:version" content="none">
```

**Simplifying guides or reference search**
We also changed the versioning of the guides and get-started by moving it outside of the versioning. That also resulted in some weird stuff because with Algolia, [we can't search for pages without certain facets](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/how-to/filter-by-null-or-missing-attributes/). By adding the explicit version of `none` to all other pages, we can simplify the [facet filter](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/in-depth/filters-and-facetfilters/) with:

```
// version:none OR version:v35.0.0
facetFilters: [['version:none', `version:${currentVersion}`]],
```

# Test Plan

Ran locally, but requires full deployment and Algolia config PR to complete this refactor/feature.

### CODE REVIEW

File: docs/components/plugins/AlgoliaSearch.js [hunk 1]
```diff
@@ -110,7 +103,8 @@ class AlgoliaSearch extends React.Component {
         return hits;
       },
       algoliaOptions: {
-        facetFilters: ignoredVersionList,
+        // include pages without version (guides/get-started) OR exact version (api-reference)
+        facetFilters: [['version:none', `version:${currentVersion}`]],
```
Comment (byCedric) [inline docs/components/plugins/AlgoliaSearch.js:107]: I'm not sure why we don't use `latest` as version too, I don't think that will cause issues. And I think we could remove the version rewrite in the handling too 😄

File: docs/components/AlgoliaDocsearchMeta.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    referenceVersion: 'latest',
+    isReferencePage: false,
+  };
+
+  getVersion() {
+    if (!this.props.isReferencePage) {
+      return 'none';
+    }
+
+    return this.props.referenceVersion === 'latest' ? LATEST_VERSION : this.props.referenceVersion;
```
Comment (byCedric) [inline docs/components/AlgoliaDocsearchMeta.js:16]: [See this comment](https://github.com/expo/expo/pull/8075/files#r417499475)

File: docs/components/plugins/AlgoliaSearch.js (hunk 1 above)
Comment (brentvatne) [reply docs/components/plugins/AlgoliaSearch.js:107]: I think we ideally don't want to have `/v37.0.0` and `/latest` both indexed

Comment (byCedric) [reply docs/components/plugins/AlgoliaSearch.js:107]: We can do that with the meta tags 😄

**Comment by byCedric** [comment]
Quickly discussed with @cruzach, I'll split this PR up into 2 so we can roll this out in 3 steps:

1. Add `<meta />` tags to our pages
2. Refactor our docsearch configuration and ask a reindex
3. Update the search logic right after the docsearch change and reminded

This is the roll-out strategy that affects search the least 😄

**Review by brentvatne — APPROVED** [review]
rollout plan sounds good to me!

**Pull request merged**